packages/hoppscotch-common/src/services/persistence/validation-schemas/index.ts

```diff
@@ -212,13 +211,23 @@ export const MQTT_REQUEST_SCHEMA = z.nullable(
 
 export const GLOBAL_ENV_SCHEMA = z.union([
   z.array(z.never()),
+
   z.array(
-    z
-      .object({
-        key: z.string(),
-        value: z.string(),
-      })
-      .strict()
+    z.union([
+      z
+        .object({
+          key: z.string(),
+          secret: z.literal(true),
+        })
+        .strict(),
+      z
+        .object({
+          key: z.string(),
+          value: z.string(),
+          secret: z.literal(false),
+        })
+        .strict(),
```

Comment:
Let's remove the strict requirement since the internal representation doesn't respect this every time.